overlord/snapstate/check_snap.go

```diff
@@ -460,6 +463,33 @@ func checkSystemUsernames(si *snap.Info) error {
 		return fmt.Errorf("system usernames are not yet supported")
 	}
 
+	// No need to check support if no system-usernames
+	if len(si.SystemUsernames) == 0 {
+		return nil
+	}
+
+	// TODO: wrap this invocation differently in a sandbox helper
+	// Run /.../snap-seccomp version-info
+	path, err := cmd.InternalToolPath("snapd")
+	if err != nil {
+		return err
+	}
+
+	vi, err := interfaces.SeccompCompilerVersionInfo(filepath.Join(filepath.Dir(path), "snap-seccomp"))
+	if err != nil {
+		return fmt.Errorf("cannot obtain seccomp compiler information: %v", err)
+	}
+
+	// If the system doesn't support robust argument filtering then we
+	// can't support system-usernames
+	err = seccomp_compiler.VersionInfo(vi).SupportsRobustArgumentFiltering()
```

Comment:
(nitpick^2) This could be a single line: `if err := seccomp_compiler.VersionInfo(vi)...; err != nil {`